### PR TITLE
[PROD-775] Update farm reward in dashboard

### PR DESCRIPTION
**Pull request by leqiang-deltafi**
1. Include farm rewards to total rewards
Screen Shot 1401-03-25 at 16 10 04
2. Only add staked farm pool to claim farm reward panel
3. made the panel wider to avoid overlap between the number and button
4. Also filter out the staked farm user for transaction
Screen Shot 1401-03-25 at 16 10 20

TODO: the total reward is calculated in Reward page and updated to dashboardView for dashboard page to display it. Need to refactor the code to make total rewards calculated with the dashboard page

### CODE REVIEW

File: src/views/Reward/Home.tsx [hunk 1]
```diff
@@ -210,10 +218,17 @@ const Home: React.FC = (props) => {
...
       return new BigNumber(0);
     }
-    return depositAmount
+
+    const rawResult = depositAmount
       .multipliedBy(apr)
       .multipliedBy(currentTs - lastUpdateTs)
       .dividedBy(SECONDS_PER_YEAR);
+
+    if (rawResult.isLessThan(exponentiatedBy("1", DELTAFI_TOKEN_DECIMALS))) {
```
Comment (leqiang-deltafi) [inline src/views/Reward/Home.tsx:227]: sometimes 'toFixed' does not floor the value. need to check if the value floored to 0

File: src/views/Reward/Home.tsx [hunk 2]
```diff
@@ -278,22 +295,26 @@ const Home: React.FC = (props) => {
...
+        anchorBnToBn(poolConfig.quoteTokenInfo, farmUser.quotePosition.depositedAmount),
       );
+
       const claimedQuoteRewards = exponentiatedBy(
         farmUser.quotePosition.cumulativeInterest.toString(),
         DELTAFI_TOKEN_DECIMALS,
       );
 
+      const untrackedRewards =
+        untrackedBaseRewards.isEqualTo(0) || untrackedQuoteRewards.isEqualTo(0)
```
Comment (leqiang-deltafi) [inline src/views/Reward/Home.tsx:312]: if one of the base/quote reward is 0, there won't be owed reward

File: src/views/Reward/Home.tsx [hunk 3]
```diff
@@ -346,6 +367,43 @@ const Home: React.FC = (props) => {
...
+          poolConfig.farmInfoList
+            .filter(
+              (farm) =>
+                !!farmPoolKeyToFarmUser[farm.farmInfo] &&
+                ((farmPoolKeyToFarmUser[farm.farmInfo].basePosition.depositedAmount.gt(new BN(0)) &&
+                  farmPoolKeyToFarmUser[farm.farmInfo].quotePosition.depositedAmount.gt(
+                    new BN(0),
+                  )) ||
+                  parseFloat(farmPoolToRewards[farm.farmInfo].unclaimedFarmRewards) > 0) &&
+                farmPoolToRewards[farm.farmInfo].totalFarmRewards !== "--",
```
Comment (leqiang-deltafi) [inline src/views/Reward/Home.tsx:399]: Select the farms that the user either has unclaimed reward or have staked shares

File: src/views/Reward/components/RewardCard.tsx [hunk 4]
```diff
@@ -138,7 +138,7 @@ const Card: React.FC<CardProps> = (props) => {
     <Box
       fontSize={{ md: 14, xs: 12 }}
       padding={{ md: 2.5, xs: 1 }}
-      minWidth={{ md: 480 }}
+      minWidth={{ md: 560 }}
```
Comment (leqiang-deltafi) [inline src/views/Reward/components/RewardCard.tsx:141]: make the pop up window wider

File: src/states/views/rewardView.ts [hunk 5]
```diff
@@ -19,6 +20,7 @@ const initialState = {
   claimResult: null,
   farmPoolToRewards: {},
   rewardRefreshTs: Math.floor(Date.now() / 1000),
+  totalDelfiRewards: new BigNumber(0),
```
Comment (ypeng-deltafi) [inline src/states/views/rewardView.ts:23]: why do we keep totalDelfiRewards in two different state?

File: src/views/Reward/Home.tsx [hunk 6]
```diff
@@ -346,6 +367,43 @@ const Home: React.FC = (props) => {
...
+  }, [totalRewardFromSwap, totalRewardFromReferral, userTotalFarmRewards, dispatch]);
+
+  const farmPoolInfoList = useMemo(
+    () =>
+      poolConfigs
+        .map((poolConfig: PoolConfig) =>
+          poolConfig.farmInfoList
+            .filter(
+              (farm) =>
+                !!farmPoolKeyToFarmUser[farm.farmInfo] &&
```
Comment (ypeng-deltafi) [inline src/views/Reward/Home.tsx:393]: this expression is very hard to review. can you simplify?

Comment (ypeng-deltafi) [reply src/views/Reward/Home.tsx:393]: can you assign `farmPoolKeyToFarmUser[farm.farmInfo]` to a variable?

File: src/states/views/rewardView.ts (hunk 5 above)
Comment (leqiang-deltafi) [reply src/states/views/rewardView.ts]: forgot to delete this field

File: src/views/Reward/Home.tsx (hunk 6 above)
Comment (ypeng-deltafi) [reply src/views/Reward/Home.tsx:393]: I think the expression will be much easy to read if we add intermediate variable for `farmPoolKeyToFarmUser[farm.farmInfo]`

Comment (leqiang-deltafi) [reply src/views/Reward/Home.tsx]: updated

File: src/views/Reward/Home.tsx [hunk 7]
```diff
@@ -458,18 +516,14 @@ const Home: React.FC = (props) => {
...
     try {
-      const farmPoolInfoList = poolConfigs
-        .map((poolConfig: PoolConfig) =>
-          poolConfig.farmInfoList
-            .filter((farm) => !!farmPoolKeyToFarmUser[farm.farmInfo])
-            .map((farm) => ({ poolConfig, farmInfo: farm.farmInfo })),
-        )
-        .flat();
+      const selectedFarmPooInfoList = farmPoolInfoList.filter(
+        ({ farmInfo }) => parseFloat(farmPoolToRewards[farmInfo].unclaimedFarmRewards) > 0,
```
Comment (ypeng-deltafi) [inline src/views/Reward/Home.tsx:525]: what is the type of unclaimedFarmRewards? why do we need parseFloat here?

Comment (leqiang-deltafi) [reply src/views/Reward/Home.tsx:525]: it is string

**Review by ypeng-deltafi — APPROVED** [review]

**Pull request merged**